app/controllers/roles_controller.rb

```diff
@@ -11,10 +11,10 @@ def create
     @role = Role.new(role_params)
     authorize @role
 
-    access_level = params[:role][:access_level].to_i
-    @role.access_level = access_level
+    plan = Plan.find(role_params[:plan_id])
+
     message = ""
-    if params[:user].present?
+    if params[:user].present? && plan.present?
```

Comment:
I would think the validations we added would remove the need for checks like this.
Were you running into issues without the check?